File: Gemini/4. Desarrollo Web Endpoint RESTful/Python/endpoint.py
from flask import Flask, request, jsonify

app = Flask(__name__)

@app.route('/saludo', methods=['GET'])
def saludo():
    nombre = request.args.get('nombre')
    if nombre:
        return jsonify({'mensaje': f'Hola, {nombre}'})
    else:
        return jsonify({'error': 'El parámetro "nombre" es requerido'}), 400

if __name__ == '__main__':
    app.run(debug=True)

File: Gemini/4. Desarrollo Web Endpoint RESTful/Golang/endpoint.go
package main

import (
	"encoding/json"
	"fmt"
	"net/http"

)

func saludoHandler(w http.ResponseWriter, r *http.Request) {
	queryParams := r.URL.Query()
	nombre := queryParams.Get("nombre")

	w.Header().Set("Content-Type", "application/json")

	if nombre != "" {
		response := map[string]string{"mensaje": fmt.Sprintf("Hola, %s", nombre)}
		json.NewEncoder(w).Encode(response)
	} else {
		w.WriteHeader(http.StatusBadRequest)
		response := map[string]string{"error": "El parámetro \"nombre\" es requerido"}
		json.NewEncoder(w).Encode(response)
	}
}

func main() {
	http.HandleFunc("/saludo", saludoHandler)
	fmt.Println("Servidor escuchando en el puerto 8080")
	http.ListenAndServe(":8080", nil)
}